### PR TITLE
Copy dependencies to requirements.txt in pre commit

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [ ] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually.
   * [ ] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [ ] I've described my changes in the section below.

### Changes description
...

### CODE REVIEW

File: .snyk/dependency-sync.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from pathlib import Path
+
+import tomllib
+
+
+def sync():
+    pyproject = tomllib.loads(Path("pyproject.toml").read_text())
+    dependencies = pyproject["project"]["dependencies"]
```
Comment (sfc-gh-mraba) [inline .snyk/dependency-sync.py:8]: Are these all dependencies including dev deps or we have some development packages listed as dependencies?  
`coverage`

Comment (sfc-gh-astus) [reply .snyk/dependency-sync.py:8]: Good point, I forgot about them

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -55,3 +55,7 @@ repos:
         pass_filenames: true
         files: ^src/.*\.py$
         exclude: ^src/snowflake/cli/app/telemetry.py$
+      - id: dependencies-sync
+        name: "Copy dependencies from pyproject.toml to requirements.txt"
+        language: system
+        entry: python .snyk/dependency-sync.py
```
Comment (sfc-gh-turbaszek) [inline .pre-commit-config.yaml:61]: Add `files` to run this only on pyproject.toml change, WDYT?

Comment (sfc-gh-astus) [reply .pre-commit-config.yaml:61]: Good idea, added

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**